squirrel/sqfuncstate.cpp

```diff
@@ -166,10 +166,10 @@ void SQFuncState::Dump(SQFunctionProto *func)
         else {
             scprintf(_SC("[%03d] %15s %d %d %d %d"), (SQInt32)n,g_InstrDesc[inst.op].name,inst._arg0,inst._arg1,inst._arg2,inst._arg3);
             switch (inst.op) {
-                case _OP_JMP: case _OP_JCMP: case _OP_JZ: case _OP_AND: case _OP_OR: case _OP_PUSHTRAP: case _OP_POSTFOREACH: case _OP_FOREACH:
+                case _OP_JMP: case _OP_JCMP: case _OP_JZ: case _OP_AND: case _OP_OR: case _OP_PUSHTRAP: case _OP_FOREACH:
                     scprintf(_SC("  jump to %d"), i + inst._arg1 + 1);
                     break;
-                case _OP_NULLCOALESCE:
+                case _OP_NULLCOALESCE: case _OP_POSTFOREACH:
```

Comment:
It seems that this debug printing is not the only issue incorrect offset (not correcting by -1).
Optimizer (SQOptimizer::optimize()) also assumes that _OP_POSTFOREACH performs jump to the instruction specified by arg1, not the previous one.
We should either fix optimizer also, or generate got differently so that it wouldn't need -1 adjustment.

Reply:
Add fix to optimizer too